src/core/QCheck.ml

```diff
@@ -1697,15 +1697,19 @@ module Test = struct
   let get_long_factor = QCheck2.Test.get_long_factor
 
   let make_cell ?if_assumptions_fail
-      ?count ?long_factor ?max_gen
+      ?count ?long_factor ?negative ?max_gen
   ?max_fail ?small:_removed_in_qcheck_2 ?retries ?name arb law
   =
   let {gen; shrink; print; collect; stats; _} = arb in
-  QCheck2.Test.make_cell_from_QCheck1 ?if_assumptions_fail ?count ?long_factor ?max_gen ?max_fail ?retries ?name ~gen ?shrink ?print ?collect ~stats law
+  QCheck2.Test.make_cell_from_QCheck1 ?if_assumptions_fail ?count ?long_factor ?negative ?max_gen ?max_fail ?retries ?name ~gen ?shrink ?print ?collect ~stats law
 
   let make ?if_assumptions_fail ?count ?long_factor ?max_gen ?max_fail ?small ?retries ?name arb law =
     QCheck2.Test.Test (make_cell ?if_assumptions_fail ?count ?long_factor ?max_gen ?max_fail ?small ?retries ?name arb law)
 
+  let make_neg ?if_assumptions_fail ?count ?long_factor ?max_gen ?max_fail ?small ?retries ?name arb law =
+    let negative = Some true in
+    QCheck2.Test.Test (make_cell ?if_assumptions_fail ?count ?long_factor ?negative ?max_gen ?max_fail ?small ?retries ?name arb law)
```

Comment:
```suggestion
    let negative = true in
    QCheck2.Test.Test (make_cell ?if_assumptions_fail ?count ?long_factor ~negative ?max_gen ?max_fail ?small ?retries ?name arb law)
```

Also that could be a good idea to have `make'` having all these parameters and then having:
```ocaml
let make = make'

let make_neg = make' ~negative:true
```

Reply:
What do you think about:
```diff
diff --git a/src/core/QCheck.ml b/src/core/QCheck.ml
index 99a05e7..04f2137 100644
--- a/src/core/QCheck.ml
+++ b/src/core/QCheck.ml
@@ -1703,12 +1703,12 @@ module Test = struct
   let {gen; shrink; print; collect; stats; _} = arb in
   QCheck2.Test.make_cell_from_QCheck1 ?if_assumptions_fail ?count ?long_factor ?negative ?max_gen ?max_fail ?retries ?name ~gen ?shrink ?print ?collect ~stats law
 
-  let make ?if_assumptions_fail ?count ?long_factor ?max_gen ?max_fail ?small ?retries ?name arb law =
-    QCheck2.Test.Test (make_cell ?if_assumptions_fail ?count ?long_factor ?max_gen ?max_fail ?small ?retries ?name arb law)
+  let make' ?if_assumptions_fail ?count ?long_factor ?max_gen ?max_fail ?small ?retries ?name ?negative arb law =
+    QCheck2.Test.Test (make_cell ?if_assumptions_fail ?count ?long_factor ?max_gen ?max_fail ?small ?retries ?name ?negative arb law)
 
-  let make_neg ?if_assumptions_fail ?count ?long_factor ?max_gen ?max_fail ?small ?retries ?name arb law =
-    let negative = true in
-    QCheck2.Test.Test (make_cell ?if_assumptions_fail ?count ?long_factor ~negative ?max_gen ?max_fail ?small ?retries ?name arb law)
+  let make = make' ~negative:false
+
+  let make_neg = make' ~negative:true
```

Reply:
Ah, thanks!
Sorry for the brain fart. I tried something along these lines and got myself more confused by the resulting type error :sweat_smile: 

I went with making `negative` a labeled (mandatory) parameter in e068041 since there are only two clients (`make` and `make_neg`) to this internal helper function - and both of them pass the parameter explcitly.